src/ucs/sys/sys.c

```diff
@@ -767,23 +767,42 @@ ucs_status_t ucs_sys_get_proc_cap(uint32_t *effective)
 #endif
 }
 
-static void ucs_sysv_shmget_error_check_EPERM(int flags, char *buf, size_t max)
+static ucs_status_t ucs_sys_get_memlock_privilege(int *lock_priv)
 {
 #if HAVE_SYS_CAPABILITY_H
-    ucs_status_t status;
     uint32_t ecap;
+    ucs_status_t status;
 
     UCS_STATIC_ASSERT(CAP_IPC_LOCK < 32);
     status = ucs_sys_get_proc_cap(&ecap);
-    if ((status == UCS_OK) && !(ecap & UCS_BIT(CAP_IPC_LOCK))) {
-        /* detected missing CAP_IPC_LOCK */
-        snprintf(buf, max, ", CAP_IPC_LOCK privilege is needed for SHM_HUGETLB");
-        return;
+    if(status != UCS_OK) {
```

Comment:
Can ignore return value (see 2 other examples)

Reply:
not sure I understand, in `ucs_sys_get_memlock_rlimit` they indeed return the status if it failed

Reply:
I meant, refactor the call to `ucs_sys_get_proc_cap` + status/cap check to a single func (see https://github.com/openucx/ucx/pull/8195/files#r867890526).

Then, instead of writing:
```
status = ucs_sys_get_proc_cap(&ecap);
if(status != UCS_OK) {
    ....
```

we can do:
```
if (!ucs_sys_has_proc_cap(CAP_IPC_LOCK)) {
  ....
```